apps/challenges/aws_utils.py

```diff
@@ -616,6 +616,112 @@ def scale_workers(queryset, num_of_tasks):
     return {"count": count, "failures": failures}
 
 
+def scale_resources(challenge, worker_cpu_cores, worker_memory):
+    """
+    The function called by scale_resources_by_challenge_pk to send the AWS ECS request to change the resources used by
+    a challenge's workers.
+
+    Deregisters the old task definition and creates a new definition that is substituted into the challenge workers.
+
+    Parameters:
+    challenge (): The challenge object for whom the task definition is being registered.
+    new_cpu_val (int): The CPU value (1 vCPU = 1024 values) that should be assigned to workers.
+    worker_memory (int): The amount of memory (MB) that should be assigned to each worker.
+
+    Returns:
+    dict: keys-> 'count': the number of workers successfully started.
+                 'failures': a dict of all the failures with their error messages and the challenge pk
+    """
+
+    from .utils import get_aws_credentials_for_challenge
+
+    client = get_boto3_client("ecs", aws_keys)
+
+    if challenge.worker_cpu_cores == worker_cpu_cores and challenge.worker_memory == worker_memory:
+        message = "Error. Worker cores and memory were not modified."
+        return {
+            "Error": message,
+            "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.BAD_REQUEST},
+        }
+
+    if not challenge.task_def_arn:
+        message = "Error. Task definition not yet registered for challenge{}.".format(
+            challenge.pk
+        )
+        return {
+            "Error": message,
+            "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.BAD_REQUEST},
+        }
+
+    # Deregister current worker service's existing task definition
+    try:
+        response = client.deregister_task_definition(
+            taskDefinition=challenge.task_def_arn
+        )
+        if (
+                response["ResponseMetadata"]["HTTPStatusCode"]
+                == HTTPStatus.OK
+        ):
+            challenge.task_def_arn = None
+            challenge.save()
+        else:
+            return response
+    except ClientError as e:
+        logger.exception(e)
+        return e.response
+
+    queue_name = challenge.queue
+    container_name = "worker_{}".format(queue_name)
+    log_group_name = get_log_group_name(challenge.pk)
+    AWS_SES_REGION_NAME = settings.AWS_SES_REGION_NAME
+    AWS_SES_REGION_ENDPOINT = settings.AWS_SES_REGION_ENDPOINT
```

Comment:
Why do we need to create new variables here? You just can just pass `settings.AWS_XXX` variable in `task_definition.format(`. Please remove these variables